addons/mail/models/ir_model.py

```diff
@@ -98,8 +98,8 @@ def _reflect_model_params(self, model):
         return vals
 
     @api.model
-    def _instanciate(self, model_data):
-        model_class = super(IrModel, self)._instanciate(model_data)
+    def __hook_post_instanciate(self, model_data, model_class):
```

Comment:
Is there a good reason to double `__` here (you don't return any value, then it shoudn't be exploitable) ? Because using `env['ir.model']._IrModel__hook_post_instanciate` becomes incorrect if the last ir_model class override doesn't have `IrModel` as a class name. Or I miss something?